config/nrfconnect/chip-module/Kconfig.features

```diff
@@ -75,18 +75,31 @@ endif # CHIP_SPI_NOR
 
 config CHIP_MEMORY_PROFILING
 	bool "Enable features for tracking memory usage"
+	# Matter stack
 	select CHIP_STATISTICS
+	# Heap
 	select CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT if CHIP_MALLOC_SYS_HEAP
-	select MBEDTLS_MEMORY_DEBUG if !CHIP_CRYPTO_PSA
-	select SYS_HEAP_RUNTIME_STATS if CHIP_MALLOC_SYS_HEAP
-	select KERNEL_SHELL
+	select SYS_HEAP_RUNTIME_STATS if CHIP_MALLOC_SYS_HEAP	
+	# Crypto
+	select MBEDTLS_MEMORY_DEBUG if CHIP_CRYPTO_MBEDTLS
+	# Network
 	select NET_STATISTICS
 	select NET_SHELL
 	select NET_BUF_POOL_USAGE
 	select OPENTHREAD_SHELL if !CHIP_WIFI
+	# Zephyr
+	select KERNEL_SHELL
 	help
 	  Enables features for tracking memory usage in Matter.
 
+# Enable full debug option when memory profiling is set.
+if CHIP_MEMORY_PROFILING
+
+config SHELL_MINIMAL
```

Comment:
We have a similar default in `Kconfig.defaults`. Which one is selected depends on whether a sample includes `Kconfig.defaults` or `Kconfig.features` first (and it seems we're not consistent wrt that). I suggest modifying `Kconfig.defaults` with proper:
```
default y if !CHIP_MEMORY_PROFILING
```
to not rely on the order of inclusion.